### ISSUE TITLE
VB -> C#: Problems with async functions

### ISSUE THREAD

**Issue by hez2010** [icsharpcode/CodeConverter#478]
### Input code
```vbnet
Namespace HLib.Tasks
    Friend Partial Module TaskExtensions
        <Extension()>
        Async Function [Then](Of T)(ByVal task As Task, ByVal f As Func(Of Task(Of T))) As Task(Of T)
            Await task
            Return Await f()
        End Function

        <Extension()>
        Async Function [Then](ByVal task As Task, ByVal f As Func(Of Task)) As Task
            Await task
            Await f()
        End Function

        <Extension()>
        Async Function [Then](Of T, U)(ByVal task As Task(Of T), ByVal f As Func(Of T, Task(Of U))) As Task(Of U)
            Return Await f(Await task)
        End Function

        <Extension()>
        Async Function [Then](Of T)(ByVal task As Task(Of T), ByVal f As Func(Of T, Task)) As Task
            Await f(Await task)
        End Function
    End Module
End Namespace
```

### Erroneous output
```C#
namespace HLib.HLib.Tasks
{
    internal static partial class TaskExtensions
    {
        public async static Task<T> Then<T>(this Task task, Func<Task<T>> f)
        {
            await task;
            return await f();
        }

        public async static Task Then(this Task task, Func<Task> f)
        {
            await task;
            await f();
            return default(Task);
        }

        public async static Task<U> Then<T, U>(this Task<T> task, Func<T, Task<U>> f)
        {
            return await f(await task);
        }

        public async static Task Then<T>(this Task<T> task, Func<T, Task> f)
        {
            await f(await task);
            return default(Task);
        }
    }
}
```

### Expected output
```C#
namespace HLib.Tasks
{
    internal static partial class TaskExtensions
    {
        public async static Task<T> Then<T>(this Task task, Func<Task<T>> f)
        {
            await task;
            return await f();
        }

        public async static Task Then(this Task task, Func<Task> f)
        {
            await task;
            await f();
        }

        public async static Task<U> Then<T, U>(this Task<T> task, Func<T, Task<U>> f)
        {
            return await f(await task);
        }

        public async static Task Then<T>(this Task<T> task, Func<T, Task> f)
        {
            await f(await task);
        }
    }
}
```

### Details
* Product in use: latest release
* Version in use: latest release

**Comment by GrahamTheCoder** [comment]
Great I think this is likely one of the issues mentored in https://github.com/icsharpcode/CodeConverter/issues/316 which I didn't previously have a repro for, thanks!

**Comment by GrahamTheCoder** [comment]
Just checked - this is still broken in latest master dadc81242b029c244db8571dbcdcfc3b389718df

**Comment by Saibamen** [comment]
Fixed in release 8.0.0
